### ISSUE TITLE
The gradient of model.output w.r.t the output of middle layer

### ISSUE THREAD

**Issue by RenShuhuai-Andy** [keras-team/keras#13499]
The model like this:
```
model = Sequential()
model.add(Dense(units=64, activation='relu', input_dim=100))
model.add(Dense(units=10, activation='relu'))
model.add(Dense(units=10, activation='softmax'))
```
I want to calculate the gradient of model.output w.r.t the output of the first layer:
```
first_layer_output = model.layers[0](model.input)
gradient = K.gradients(model.output, first_layer_output)
```
but the result of `gradient` is `None`.

When I use `first_layer_output = model.get_layer(index=0).output`, it works. 

Anything different between this two ways? The gradient of model.output can't backpropagate to the `first_layer_output` in the second way?
